packages/apps/shopify-app-remix/src/server/authenticate/public/extension/authenticate.ts

```diff
@@ -0,0 +1,50 @@
+import {BasicParams} from 'src/server/types';
+
+import {
+  respondToBotRequest,
+  respondToOptionsRequest,
+  getSessionTokenHeader,
+  validateSessionToken,
+  ensureCORSHeadersFactory,
+} from '../../helpers';
+
+import {AuthenticateExtension, ExtensionContext} from './types';
+
+export function authenticateExtensionFactory(
```

Comment:
I wonder if we can touch up the types a little bit so:
- we don't have this little bit of duplication
- we don't have to cast the types

Maybe we can have something like:

```suggestion
export function authenticateExtensionFactory<Options, Context>(): (Request, Options) => Promise<Context> { ... }
```

that way we wouldn't need the generic empty types, and the types returned by this function could be used without casting.